spring-batch-core/src/main/java/org/springframework/batch/core/step/AbstractStep.java

```diff
@@ -256,6 +261,13 @@ public final void execute(StepExecution stepExecution) throws JobInterruptedExce
 						+ "This job is now in an unknown state and should not be restarted.", name, stepExecution.getJobExecution().getJobInstance().getJobName()), e);
 			}
 
+			sample.stop(Timer.builder(METRICS_PREFIX + "step")
+							.description("Step duration in seconds")
```

Comment:
See above comment about "seconds".

Reply:
Make sense. Removed the unit from all descriptions.